modules/renter/contractor/formcontract.go

```diff
@@ -28,34 +28,47 @@ var (
 
 // maxSectors is the estimated maximum number of sectors that the allowance
 // can support.
-func maxSectors(a modules.Allowance, hdb hostDB) (uint64, error) {
-	if a.Hosts == 0 || a.Period == 0 {
+func maxSectors(a modules.Allowance, hdb hostDB, tp transactionPool) (uint64, error) {
+	if a.Hosts <= 0 || a.Period <= 0 {
 		return 0, errors.New("invalid allowance")
 	}
 
 	// Sample at least 10 hosts.
 	nRandomHosts := int(a.Hosts)
-	if nRandomHosts < 10 {
-		nRandomHosts = 10
+	if nRandomHosts < minHostsForEstimations {
+		nRandomHosts = minHostsForEstimations
 	}
 	hosts := hdb.RandomHosts(nRandomHosts, nil)
 	if len(hosts) < int(a.Hosts) {
 		return 0, fmt.Errorf("not enough hosts in hostdb for sector calculation, got %v but needed %v", len(hosts), int(a.Hosts))
 	}
 
-	// Calculate cost of storing 1 sector per host for the allowance period.
-	var sum types.Currency
+	// Calculate cost of creating contracts with each host, and the cost of
+	// storing sectors on each host.
+	var sectorSum types.Currency
+	var contractCostSum types.Currency
 	for _, h := range hosts {
-		sum = sum.Add(h.StoragePrice)
+		sectorSum = sectorSum.Add(h.StoragePrice)
+		contractCostSum = contractCostSum.Add(h.ContractPrice)
+	}
+	averageSectorPrice := sectorSum.Div64(uint64(len(hosts)))
+	averageContractPrice := contractCostSum.Div64(uint64(len(hosts)))
+	costPerSector := averageSectorPrice.Mul64(a.Hosts).Mul64(modules.SectorSize).Mul64(uint64(a.Period))
+	costForContracts := averageContractPrice.Mul64(a.Hosts)
+
+	// Subtract fees for creating the file contracts from the allowance.
+	_, feeEstimation := tp.FeeEstimation()
+	costForTxnFees := types.NewCurrency64(estimatedFileContractTransactionSize).Mul(feeEstimation).Mul64(a.Hosts)
+	// Check for potential divide by zero
+	if a.Funds.Cmp(costForTxnFees.Add(costForContracts)) < 0 {
```

Comment:
If a.funds == cost+fee, you'll get a divide by zero later because you subtract them.

I think the average user would rather not spend more on fees than on actual storage costs, but it's hard to apply that sort of rule to everyone. In my ideal client, there would be a popup slider thing where you could adjust either the desired cost or the desired storage, and it would also highlight how much was being spent on fees.